src/middlewares/error_handlers.cr

```diff
@@ -1,16 +1,76 @@
+require "http"
+require "json"
+
+struct ErrorData
+  include JSON::Serializable
+  property type : String
+  property message : String?
+  property status_code : HTTP::Status?
+
+  def initialize(@type : String,
+                 @message : String?,
+                 @status_code : HTTP::Status? = nil)
+  end
+
+  def self.new(error : Exception)
+    if error.is_a? DppmRestApi::HTTPStatusError
```

Comment:
This doesn't work as a method overload (`def self.new(error : DppmRestApi::HTTPStatusError)`)?

Reply:
Yeah, I just double-checked and indeed that does not work because `DppmRestApi::HTTPStatusError` inherits from `Exception`